client/src/commonMain/kotlin/com/algolia/search/model/insights/ObjectData.kt

```diff
@@ -0,0 +1,57 @@
+import com.algolia.search.serialize.internal.Key
+import com.algolia.search.serialize.internal.asJsonInput
+import com.algolia.search.serialize.internal.asJsonOutput
+import com.algolia.search.serialize.internal.jsonPrimitiveOrNull
+import kotlinx.serialization.*
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+/**
+ * ObjectData
+ *
+ * @param queryID Unique identifier for a search query, used to track purchase events with multiple records that originate from different searches.
+ * @param price
+ * @param quantity Quantity of a product that has been purchased or added to the cart. The total purchase value is the sum of `quantity` multiplied with the `price` for each purchased item.
+ * @param discount
+ */
+@Serializable(ObjectData.Companion::class)
+public data class ObjectData(
+
+    /** Unique identifier for a search query, used to track purchase events with multiple records that originate from different searches. */
+    @SerialName(value = Key.QueryID) val queryID: String? = null,
```

Comment:
For consistency with other parts of the codebase, this should be `QueryID` instead of `String`